### PR TITLE
MEPTS-240: Validate partner metadata

### PR DESCRIPTION
**Pull request by ynurmahomed**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -34,6 +35,7 @@ dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-web'
 	implementation 'pentaho-kettle:kettle-core:8.2.0.7-719'
 	implementation 'pentaho-kettle:kettle-engine:8.2.0.7-719'
+	pdiPlugin 'org.pentaho.di.plugins:pdi-core-plugins-impl:8.2.0.7-719'
```
Comment (kaweesi) [inline build.gradle:38]: this seems to be available from maven, https://javalibs.com/artifact/org.pentaho.di.plugins/pdi-core-plugins-impl is this the way u load it from there!

File: build.gradle [hunk 2]
```diff
@@ -58,4 +60,11 @@ jacocoTestReport {
 		xml.enabled = true
 		html.enabled = true
 	}
-}
\ No newline at end of file
+}
+
+task copyPlugins(type: Copy) {
+	from configurations.pdiPlugin
```
Comment (kaweesi) [inline build.gradle:67]: BTW what reason is behind copying it into `plugins/`?

File: src/main/java/com/openmrs/migrator/MigratorApplication.java [hunk 3]
```diff
@@ -100,6 +101,7 @@ private void executeSetupCommand() throws IOException {
     pdiFiles.add("pdiresources/transformations/validate-visit-attribute-types.ktr");
     pdiFiles.add("pdiresources/transformations/validate-visit-types.ktr");
     pdiFiles.add("settings.properties");
+    pdiFiles.add("plugins/pdi-core-plugins-impl-8.2.0.7-719.jar");
```
Comment (kaweesi) [inline src/main/java/com/openmrs/migrator/MigratorApplication.java]: what are we planning to use this folder for?

File: src/main/resources/pdiresources/transformations/merge-location.ktr [hunk 4]
```diff
@@ -1,823 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kaweesi) [inline src/main/resources/pdiresources/transformations/merge-location.ktr:1]: why have you decided to get rid of this please?

**Review by kaweesi — COMMENTED** [review]
thanks a lot @ynurmahomed for the great work, i have left some few comments please

File: src/main/java/com/openmrs/migrator/MigratorApplication.java [hunk 5]
```diff
@@ -35,7 +35,8 @@
           "config/",
           "pdiresources/",
           "pdiresources/transformations/",
-          "pdiresources/jobs/");
+          "pdiresources/jobs/",
+          "plugins/");
```
Comment (kaweesi) [inline src/main/java/com/openmrs/migrator/MigratorApplication.java]: a lot has changed in this file, pull from develop please! these changes are now meant to go into Migrator.java

Comment (ynurmahomed) [reply src/main/java/com/openmrs/migrator/MigratorApplication.java]: :+1:

File: src/main/resources/pdiresources/transformations/merge-location.ktr (hunk 4 above)
Comment (ynurmahomed) [reply src/main/resources/pdiresources/transformations/merge-location.ktr:1]: This was added by mistake. The transformation was not complete, but file is still available in the corresponding ticket.

File: build.gradle (hunk 1 above)
Comment (ynurmahomed) [reply build.gradle:38]: I tried to explain it on 36fa4cf. 

`kettle-core` does not include all types of steps that you can use in `spoon`. 

Some steps are defined in plugins that are loaded at runtime by kettle. It seems that the only way to load this plugins is to store them in a folder that kettle will scan when starting (no classpath scanning).

[You can either use `./plugins`, `~/.kettle/plugins` or define system property `KETTLE_PLUGIN_BASE_FOLDERS`](https://github.com/pentaho/pentaho-kettle/blob/0cd5339daeca9baafb72c26ea1ea5dd98174e884/core/src/main/java/org/pentaho/di/core/plugins/PluginFolder.java#L97-L100).

The [Abort](https://github.com/pentaho/pentaho-kettle/blob/8.2.0.7-R/plugins/core/impl/src/main/java/org/pentaho/di/trans/steps/abort/Abort.java) used in the validations transformation is not included in core.

I think the best way to handle this situation is to have a custom gradle configuration for separating the scope of this plugin dependencies and to rely on `./plugins` to have kettle load them. That is the reason for `pdiPlugin`, the copy step before build and extracting the jar to `plugins/pdi-core-plugins-impl-8.2.0.7-719.jar`.

Maybe there is a better way to handle this.

**Pull request merged**